Makefile

```diff
@@ -403,6 +403,15 @@ distclean: | clean
 
 dataclean:
 	rm -fR data/*.tar.gz data/*.bin data/data_ok
+	
+alpakaclean:
+	rm -fR lib/alpaka obj/alpaka test/alpaka alpaka
+	
+cudaclean:
+	rm -fR lib/cuda obj/cuda test/cuda cuda
+	
+kokkosclean:
+	rm -fR lib/kokkos obj/kokkos test/kokkos kokkos	
```

Comment:
Is this level of precision for cleaning useful? (in my use cases cleaning everything has been sufficient)

If it is, this would be better done in a general way.

Reply:
Yes, find it nice to be able to clean for example src/alpaka, without having to recompile all the rest.
But yes will push a more generic way

Reply:
That would be along
```Makefile
define CLEAN_template
clean_$(1):
	rm -fR lib/$(1) obj/$(1) test/$(1) $(1)
endef
$(foreach target,$(TARGETS_ALL),$(eval $(call CLEAN_template,$(target))))
```

Reply:
Ok I put your function there, thanks